administration/src/mui-modules/base/__tests__/BaseMenu.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { EditNote, PrintOutlined } from '@mui/icons-material'
+import { fireEvent, render } from '@testing-library/react'
+import React from 'react'
+
+import BaseMenu from '../BaseMenu'
+
+describe('BaseMenu', () => {
+  const mockMenuItems = [
+    {
+      name: 'Print PDF',
+      onClick: jest.fn(),
+      icon: <PrintOutlined />,
+    },
+    {
+      name: 'Edit Note',
+      onClick: jest.fn(),
+      icon: <EditNote />,
+    },
+  ]
+
+  const defaultProps = {
+    menuLabel: 'Test Menu',
+    menuItems: mockMenuItems,
+    containerWidth: 200,
+    itemHeight: 48,
+  }
+
+  it('should render menu button with correct label', () => {
+    const { getByRole } = render(<BaseMenu {...defaultProps} />)
+    const menuButton = getByRole('button', { name: 'Test Menu' })
+
+    expect(menuButton).toBeTruthy()
+  })
+
+  it('should open menu when button is clicked', () => {
+    const { getByRole } = render(<BaseMenu {...defaultProps} />)
+    const menuButton = getByRole('button', { name: 'Test Menu' })
+
+    fireEvent.click(menuButton)
+    const menu = getByRole('menu')
+
+    expect(menu).toBeTruthy()
+  })
+
+  it('should render all menu items when the menu is open', () => {
+    const { getByRole, getByText } = render(<BaseMenu {...defaultProps} />)
+    const menuButton = getByRole('button', { name: 'Test Menu' })
+
+    fireEvent.click(menuButton)
+
+    expect(getByText(mockMenuItems[0].name)).toBeTruthy()
+    expect(getByText(mockMenuItems[1].name)).toBeTruthy()
+  })
+
+  it('should call onClick handler when menu item is clicked', () => {
+    const { getByRole, getByText } = render(<BaseMenu {...defaultProps} />)
+    const menuButton = getByRole('button', { name: 'Test Menu' })
+
+    fireEvent.click(menuButton)
+    const menuItem = getByText('Print PDF')
+    fireEvent.click(menuItem)
+
+    expect(mockMenuItems[0].onClick).toHaveBeenCalledTimes(1)
+  })
+
+  it('should close menu after clicking menu item', () => {
+    const { getByRole, getByText, queryByRole } = render(<BaseMenu {...defaultProps} />)
+    const menuButton = getByRole('button', { name: 'Test Menu' })
+
+    fireEvent.click(menuButton)
+    const menuItem = getByText('Print PDF')
+    fireEvent.click(menuItem)
+
+    const menu = queryByRole('menu')
+    expect(menu).toBeNull()
+  })
+})
```

Comment:
If I put the menu expect statement before clicking on menuItem it will get a pass 🫣.
something wrong here: ex.
```
  it('should close menu after clicking menu item', () => {
    const { getByRole, getByText, queryByRole } = render(<BaseMenu {...defaultProps} />)
    const menu = queryByRole('menu')

    expect(menu).toBeNull()
    const menuButton = getByRole('button', { name: 'Test Menu' })

    fireEvent.click(menuButton)
    expect(menu).toBeNull()
    const menuItem = getByText('Print PDF')
    fireEvent.click(menuItem)
  })
```

Reply:
Well i dunno why this is the case but if you move the menu item before the click menu button, it will not be found (which is correct). Not sure what happens in mui under the hood. Maybe the menu is in the dom but not visible.
The important thing is that the items are not visible before clicking the "menuButton"

Reply:
Yes i guess this is due to this is inside a modal and its always difficult to check that.
I spent some time in investigating and even with the suggestion here:
https://stackoverflow.com/questions/69889380/how-to-test-material-ui-popover-close-implementation
it doesn't work.
If you want i can remove the test. The component actually works and i won't spend more time in unit testing here

Reply:
I will give it a look again if I can find a solution

Reply:
I think you need to apply the same for "should call onClick handler when menu item is clicked" test.